python3/httplib2/__init__.py

```diff
@@ -151,15 +151,21 @@ def _build_ssl_context(
     # source: https://docs.python.org/3/library/ssl.html#ssl.SSLContext.maximum_version
     if maximum_version is not None:
         if hasattr(context, "maximum_version"):
-            context.maximum_version = getattr(ssl.TLSVersion, maximum_version)
+            if isinstance(maximum_version, str):
+                context.maximum_version = getattr(ssl.TLSVersion, maximum_version)
+            else:
+                context.maximum_version = maximum_version
```

Comment:
In general, great idea, thanks. What do you think on this approach?

```suggestion
            if not isinstance(maximum_version, ssl.TLSVersion):
                maximum_version = getattr(ssl.TLSVersion, maximum_version)
            context.maximum_version = maximum_version
```

Reply:
Even better! Thanks for suggesting, I'll fix it up.

Reply:
I've kept the test focusing on only calling getattr is it in a str since that's all that getattr accepts. If someone passes a bad enum to us, we'll pass it upstream and let that crash if it must but then developers looking at the error message will look at Python docs and realise it should've been the TLSVersion enum. LMK if you're prefer to handle it differently.

Reply:
Just tried it and context setattr checks enum value so your version is more helpful.